### DISCUSSION TITLE
Using "Is Full Record Set" when bulk update of children means "Child Object Where Clause" returns 0 records

### DISCUSSION THREAD

**Discussion by louisskelton** [jamessimone/apex-rollup#488] [Q&A]
Hi! So excited to find this. What a fantastic project.

I've been having some issues with some rollups (migrated from DLRS) which I've been trying to diagnose. I think the issue results from the following:

1. Using **Is Full Record Set** on a SUM rollup
2. Using a **Child Object Where Clause** (in this case `Status__c != 'Void'`)
3. Apex triggering the rollup following performing a bulk update of the child records such that following the update no children match the  **Child Object Where Clause** criteria (e.g in this case updating all to Status__c = `Void`).

I end up getting a negative version of the full SUM rather than the expected `0` or some other incorrect value (always negative though).

I can fix this by adding 'Status__c' to **Changed Fields On Child Object** (though the Help for that field says it shouldn't be used alongside **Child Object Where Clause**).

So my questions are:
1. Have I correctly diagnosed the issue?
2. Is this the correct solution? Ideally I'd rather not have to look at each rollup and try and cover off this particular scenario so I'm hoping there's a setting I've missed.

Thank you so much

Louis

**Comment by jamessimone** [comment]
@louisskelton Thanks for reaching out - I will take a look! When you say "a negative version of the full SUM", I'm assuming you mean "the value it should be, but negative" ? Just looking to confirm - thanks!

**Reply by louisskelton** [reply]
Yes that’s it! I found the same result when using both queueable and synchronous modes. One thing I forgot to mention - the child records are themselves being updated in a batch following an update to the parent record (ie - set status of Parent to ‘Void’ triggers batch update of all child records). I’ll test tomorrow whether updating in real-time vs batch makes a difference. The batch size exceeds the number of child records, I should add…

**Reply by louisskelton** [reply]
Just ran that test - I get the same issue regardless of whether the process that updates all the child records occurs in batch or realtime. Thanks for taking a look :)

**Reply by jamessimone** [reply]
Thanks for doing that! I appreciate the additional context

**Reply by jamessimone** [reply]
@louisskelton I've been able to reproduce this issue - and can confirm the workaround with the `Changed Fields On Child Object` . Obviously this isn't ideal as a long-term solution, nor do I intend it to be. I've been wrapping up a few other changes for the next version, but I'm hopeful to include the fix for this in there! In the meantime, your workaround will continue to work.

**Reply by jamessimone** [reply]
@louisskelton looking for a bit more info as I get down to fixing this issue - how are you triggering the full recalc of the children when the parent record is getting updated? Just looking to make sure I'm covering all the possible ways this can happen - thank you!

**Reply by jamessimone** [reply]
Ah wait, I must have missed your comment about the batch process triggering the update of all child records. Second, and probably more pertinent question - how many children records are we talking about getting updated here?

**Reply by louisskelton** [reply]
Thanks @jamessimone! In the example I was testing it was 81 child records. I've tested both realtime (i.e update all of them) and batch (e.g 20 at a time). We have a custom setting that determines when to use realtime and the size of the batch in it exceeds the realtime limit. Because we're able to process in batch I'd hope to be able to update many records at once... but practically speaking I'd hope it'd never be more than 200...

**Reply by jamessimone** [reply]
That shouldn't be a problem, regardless. I've fixed this issue locally and am hoping to release the updated version later today.

**Reply by louisskelton** [reply]
Just tested - fix works great! Thanks so much for this quick response! 👌 I'll yell if I find anything else.

**Reply by jamessimone** [reply]
Perfect. Glad to hear it!
